From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "A. Cody Schuffelen" <schuffelen@google.com>
Date: Tue, 11 Feb 2020 18:01:10 -0800
Subject: ANDROID: Add TPM support and the vTPM proxy to Cuttlefish.

This module allows presenting the kernel TPM interface while proxying
the TPM commands into a file descriptor. The module was originally
implemented to support running a TPM simulator on the same host system
and exposing a kernel TPM interface to a Linux container, but it is also
a convenient incremental step while we figure out our long-term strategy
with crosvm, which does not have TPM support following the same
standards as qemu.

CONFIG_TCG_TPM, the base config for the various TPM drivers, required
CONFIG_SECURITYFS. CONFIG_SECURITYFS exists only as a boolean and not a
tristate, so we can't install it as a module.

Bug: 148102533
Test: Build and run locally with cuttlefish, check for /dev/vtpmx
Change-Id: I568a50c2ecb7899aae70e7a20efaedc84443511d
Signed-off-by: A. Cody Schuffelen <schuffelen@google.com>
[joneslee: Removed defconfig elements]
Signed-off-by: Lee Jones <lee.jones@linaro.org>
---
 cuttlefish.fragment | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/cuttlefish.fragment b/cuttlefish.fragment
--- a/cuttlefish.fragment
+++ b/cuttlefish.fragment
@@ -18,3 +18,5 @@ CONFIG_VIRTIO_MMIO_CMDLINE_DEVICES=y
 CONFIG_INCREMENTAL_FS=m
 CONFIG_TEST_STACKINIT=m
 CONFIG_TEST_MEMINIT=m
+CONFIG_TCG_TPM=m
+CONFIG_TCG_VTPM_PROXY=m
